### PR TITLE
An option to fix DI container dependency

### LINKED ISSUE #44: Missing container dependency
`bin/yii` file is [using our own DI container](https://github.com/yiisoft/yii-console/blob/master/bin/yii#L5). I guess we shouldn't require it, but we should not use it directly. Any suggestions?

### PR DESCRIPTION
**Pull request by viktorprogger**
A way to fix #44.

### CODE REVIEW

**Comment by viktorprogger** [comment]
Is it ok that I can't set "ready for review" label (and other labels too)?

**Comment by samdark** [comment]
> Is it ok that I can't set "ready for review" label (and other labels too)?

Reviewers group was missing from this repository. Fixed.

File: composer.json [hunk 1]
```diff
@@ -26,6 +26,9 @@
     "require-dev": {
         "phpunit/phpunit": "^8.4"
     },
+    "suggest": {
```
Comment (roxblnfk) [inline composer.json]: Add `psr/container` to `require` section

File: bin/yii [hunk 2]
```diff
@@ -37,7 +37,10 @@ use Yiisoft\Yii\Console\Application;
     }
     require_once $autoloadPath;
 
-    $container = new Container(require Builder::path('console'));
+    $params = require Builder::path('params');
+
+    /** @var ContainerInterface $container */
+    $container = $params['container']();
```
Comment (xepozz) [inline bin/yii]: How about to use full class name as a key?

File: config/params.php [hunk 3]
```diff
@@ -6,4 +6,13 @@
     'commands' => [
         'serve' => \Yiisoft\Yii\Console\Command\Serve::class,
     ],
+    'container' => static function (): Yiisoft\Di\Container {
+        if (!class_exists('Yiisoft\Di\Container')) {
+            $message = 'You must either install yiisoft/di or create "container" configuration parameter'
+                . ' containing a callable which will return configured implementation of PSR-11 DI container.';
+            throw new \hiqdev\composer\config\exceptions\BadConfigurationException($message);
```
Comment (xepozz) [inline config/params.php]: Why did you use this class?

File: bin/yii (hunk 2 above)
Comment (roxblnfk) [reply bin/yii]: Or `console.container`

File: README.md [hunk 4]
```diff
@@ -10,6 +10,13 @@
...
 This [Yii Framework] extension allows easy creation of console applications.
 
+Main usage: `vendor/bin/yii <command>`
+
+To start working with it you must do one of these:
+- `composer require yiisoft/di`
+- Manually create a `container` option in `params` (see [Composer config plugin](https://github.com/hiqdev/composer-config-plugin) to get more info)
+- Use your own binary file instead of `vendor/bin/yii`
+
 [Yii Framework]: https://github.com/yiisoft/core
```
Comment (rustamwin) [inline README.md]: This link is already out of date

File: composer.json (hunk 1 above)
Comment (samdark) [reply composer.json]: `psr/container-implementation` should be added as well since console can't work w/o container.

File: bin/yii (hunk 2 above)
Comment (samdark) [reply bin/yii]: `console.container` sounds good.

File: README.md (hunk 4 above)
Comment (samdark) [reply README.md]: It is.

**Pull request merged**